nifi-nar-bundles/nifi-azure-bundle/nifi-azure-reporting-task/src/main/java/org/apache/nifi/reporting/azure/loganalytics/AzureLogAnalyticsReportingTask.java

```diff
@@ -0,0 +1,335 @@
+            .name("Process group ID(s)")
+            .description("If specified, the reporting task will send metrics the configured ProcessGroup(s) only. Multiple IDs should be separated by a comma. If"
+                    + " none of the group-IDs could be found or no IDs are defined, the NiFi-Flow-ProcessGroup is used and global metrics are sent.")
+            .required(false)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .addValidator(StandardValidators
+                    .createListValidator(true, true, StandardValidators.createRegexMatchingValidator(Pattern.compile("[0-9a-z-]+"))))
+            .build();
+    static final PropertyDescriptor JOB_NAME = new PropertyDescriptor.Builder()
+            .name("The job name")
+            .description("The name of the exporting job")
+            .defaultValue("nifi_reporting_job")
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+    static final PropertyDescriptor SEND_JVM_METRICS = new PropertyDescriptor.Builder()
+            .name("Send JVM-metrics")
+            .description("Send JVM-metrics in addition to the NiFi-metrics")
+            .allowableValues("true", "false")
+            .defaultValue("false")
+            .required(true)
+            .build();
+    static final PropertyDescriptor LOG_ANALYTICS_URL_ENDPOINT_FORMAT = new PropertyDescriptor.Builder()
+            .name("Log Analytics URL Endpoint Format")
+            .description("Log Analytics URL Endpoint Format")
+            .required(false)
+            .defaultValue("https://{0}.ods.opinsights.azure.com/api/logs?api-version=2016-04-01")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .build();
+
+    private String createAuthorization(String workspaceId, String key, int contentLength, String rfc1123Date) {
+        try {
+            // Documentation: https://docs.microsoft.com/en-us/rest/api/loganalytics/create-request
+            String signature = String.format("POST\n%d\napplication/json\nx-ms-date:%s\n/api/logs", contentLength, rfc1123Date);
+            Mac mac = Mac.getInstance(HMAC_SHA256_ALG);
+            mac.init(new SecretKeySpec(DatatypeConverter.parseBase64Binary(key), HMAC_SHA256_ALG));
+            String hmac = DatatypeConverter.printBase64Binary(mac.doFinal(signature.getBytes(UTF8)));
+            return String.format("SharedKey %s:%s", workspaceId, hmac);
+        } catch (NoSuchAlgorithmException | InvalidKeyException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @OnScheduled
+    public void setup(final ConfigurationContext context) {
+       virtualMachineMetrics = JmxJvmMetrics.getInstance();
```

Comment:
It seems to make sense to get rid of this method and instead just declare `virtualMachineMetrics` as final and instantiate in the declaration.

Reply:
fixed as recommended